pkg/docker/creds/credentials.go

```diff
@@ -23,9 +23,10 @@ import (
 	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
 
 	"knative.dev/func/pkg/docker"
+	fn "knative.dev/func/pkg/functions"
 )
 
-type CredentialsCallback func(registry string) (docker.Credentials, error)
+type CredentialsCallback func(f fn.Function, registry string) (docker.Credentials, error)
```

Comment:
@gauron99  I do not see any reason for adding `fn.Function` here. The `registry` parameter is completely sufficient.

Reply:
Just use registry at https://github.com/knative/func/pull/2596/files#diff-753067f7ed9e5f834cd3d8219369a40264abf6ade1665a12e680430f3986c675R27

Reply:
```go
fmt.Fprintf(out, "Incorrect credentials for registry '%s'. Please make sure the registry is correct and try again.\n", registry)
```

Reply:
it is so that I can provide the image name and print it out, making it more obvious what are the credentials "used for"

Reply:
Hmm I thought that registry also included project/namespace something like `docker.io/johndoe`.

Reply:
I will clean it up a bit, wait. I will ping you when its ready